middleware/kubernetes/kubernetes.go

```diff
@@ -45,15 +45,39 @@ type Kubernetes struct {
 	Selector      *labels.Selector
 }
 
+type endpoint struct {
+	addr api.EndpointAddress
+	port api.EndpointPort
+}
+
+type service struct {
+	name      string
+	namespace string
+	addr      string
+	ports     []api.ServicePort
+	endpoints []endpoint
+}
+
 var errNoItems = errors.New("no items found")
 var errNsNotExposed = errors.New("namespace is not exposed")
+var errInvalidRequest = errors.New("invalid query name")
 
 // Services implements the ServiceBackend interface.
 func (k *Kubernetes) Services(state request.Request, exact bool, opt middleware.Options) ([]msg.Service, []msg.Service, error) {
+
+	if state.Type() == "SRV" && !validSrvPrefix(state.QName()) {
+		return nil, nil, errInvalidRequest
+	}
+
 	s, e := k.Records(state.Name(), exact)
 	return s, nil, e // Haven't implemented debug queries yet.
 }
 
+func validSrvPrefix(name string) bool {
+	matched, _ := regexp.MatchString("^_.*\\._.*\\..*", name)
```

Comment:
Practically speaking, yes in the sense that kubernetes only serves tcp and udp. However the RFC says ...

>... _TCP and _UDP are at present the most useful values for this field, though any name defined by Assigned Numbers or locally may be used (as for Service). ...

That said - its more efficient to short circuit and return an error before searching fruitlessly to produce the same result.